ui/src/components/shared/ActionButtons.vue

```diff
@@ -1,88 +1,75 @@
 <template>
-  <Button
-    v-if="show('runQuery')"
-    :icon="'fa-solid fa-bolt'"
-    :severity="getSeverity()"
-    :class="getClass()"
-    @click="onRunQuery(iri)"
-    v-tooltip.top="'Run query'"
-    data-testid="run-query-button"
-  />
-  <Button
-    v-if="show('findInTree')"
-    :icon="fontAwesomePro ? 'fa-duotone fa-list-tree' : 'fa-solid fa-sitemap'"
-    :severity="getSeverity()"
-    :class="getClass()"
-    @click="locateInTree(iri)"
-    v-tooltip.top="'Find in tree'"
-    data-testid="select-button"
-  />
-  <Button
-    v-if="show('view')"
-    :icon="fontAwesomePro ? 'fa-duotone fa-up-right-from-square' : 'fa-solid fa-up-right-from-square'"
-    :severity="getSeverity()"
-    :class="getClass()"
-    @click="directService.view(iri)"
-    v-tooltip.top="'View'"
-    data-testid="view-button"
-  />
-  <Button
-    v-if="show('edit')"
-    :icon="fontAwesomePro ? 'fa-duotone fa-pen-to-square' : 'fa-solid fa-pen-to-square'"
-    :severity="getSeverity()"
-    :class="getClass()"
-    @click="directService.edit(iri)"
-    v-tooltip.top="'Edit'"
-    data-testid="edit-button"
-  />
-  <Button
-    v-if="show('favourite') && isFavourite(iri)"
-    style="color: var(--yellow-500)"
-    icon="fa-solid fa-star"
-    :severity="getSeverity()"
-    :class="getClass()"
-    @click="updateFavourites(iri)"
-    v-tooltip.left="'Unfavourite'"
-    data-testid="unfavourite-button"
-  />
-  <Button
-    v-else-if="show('favourite') && !isFavourite(iri)"
-    icon="fa-regular fa-star"
-    :severity="getSeverity()"
-    :class="getClass()"
-    @click="updateFavourites(iri)"
-    v-tooltip.left="'Favourite'"
-    data-testid="favourite-button"
-  />
-  <TestQueryParams
-    v-if="showTestQueryParams && isObjectHasKeys(queryRequest)"
-    :showDialog="showTestQueryParams"
-    :query-request="queryRequest"
-    :params="params"
-    @on-params-populated="onParamsPopulated"
-    @close-dialog="showTestQueryParams = false"
-  />
-  <TestQueryResults
-    v-if="showTestQueryResults && isObjectHasKeys(queryRequest)"
-    :showDialog="showTestQueryResults"
-    :query-request="queryRequest"
-    @close-dialog="showTestQueryResults = false"
-  />
+  <div class="flex">
+    <QuickQuery v-if="show('quickQuery')" :query-iri="iri">
+      <template #button="{ runQuickQuery }">
+        <Button
+          :icon="'fa-solid fa-bolt'"
+          :severity="getSeverity()"
```

Comment:
tip: only need function run brackets if passing parameters. Both work though.

:severity="getSeverity"

Reply:
There is a difference in event handling between :severity="getSeverity" and :severity="getSeverity()". In this case the function needs the brackets to be invoked. 
The only case where the use of brackets is optional is when the function is tied to a specific event e.g. @click - @click="getSeverity()" === @click="getSeverity".
I prefer to always use brackets for consistency.